Libraries/Interface/navigator.cpp

```diff
@@ -0,0 +1,73 @@
+/**
+ * Adds selectable widget to the end of selectable widgets queue
+ * @param {Widget*} unit           - seleclable widget
+ * @param {int }    subunitsAmount - amount of subwidgets
+ */
+void Navigator::pushUnit(Widget* unit, int subunitsAmount)
+{
+    units.push_back({unit, subunitsAmount});
+}
+
+/**
+ * Starts main loop of application to ctch selections and activate events done by user
+ */
+void Navigator::listen()
+{
+    curs_set(0);
+    noecho();
+    currentUnit = units.begin();
+    currentSubunit = 0;
+    currentUnit->widget->focus(currentSubunit);
+
+    while (1)
+    {
+        int command = getch();
+
+        if (command != ERR)
+        {
+            switch(command)
+            {
+                case (119):
```

Comment:
Move from 'wasd' control scheme to arrows using the keypad(stdscr, true); and ncurses-defined macro (KEY_UP, etc)